airflow/models/param.py

```diff
@@ -49,6 +49,7 @@ class Param:
     """
 
     __NO_VALUE_SENTINEL = NoValueSentinel()
+    CLASS_IDENTIFIER = '__class'
```

Comment:
In the original PR, I initially used `__type` only, but Ash felt that it should be called `__class`, cause it would be representing a Param class & not a native type, which made sense to me.